modules/ssh/ssh.go

```diff
@@ -266,12 +268,23 @@ func NoOpHostKeyCallback(hostname string, remote net.Addr, key ssh.PublicKey) er
 }
 
 func createAuthMethodsForHost(host Host) ([]ssh.AuthMethod, error) {
-	signer, err := ssh.ParsePrivateKey([]byte(host.SshKeyPair.PrivateKey))
-	if err != nil {
-		return []ssh.AuthMethod{}, err
+	var methods []ssh.AuthMethod
+	var err error
+	if host.SshAgent {
+		socket := os.Getenv("SSH_AUTH_SOCK")
+		conn, errDial := net.Dial("unix", socket)
+		if errDial == nil {
```

Comment:
Shouldn't you return an `errDial` if it's not `nil`? Otherwise, the error is silently swallowed...

Reply:
Perhaps I'd return `errDial` in case no SshKeyPair was provided. If both SshAgent and SshKeyPair are enabled, I'd return an error only if both fail. What do you think? Any error returned by `createAuthMethodsForHost` would make functions using it immediately cascade the failure.

Reply:
I suspect swallowing the error will be more surprising/problematic for users that set `SshAgent` to `true`, so I think it should be returned/cascaded immediately.